scipy/stats/_sampling.py

```diff
@@ -1051,23 +1043,15 @@ def evaluate_error(self, size=100000, random_state=None, x_error=False):
         """
         if not isinstance(size, (numbers.Integral, np.integer)):
             raise ValueError("size must be an integer.")
-        if random_state is None:
-            urng = self.random_state
-        else:
-            urng = check_random_state(random_state)
-        if isinstance(urng, np.random.RandomState):
-            state = urng.get_state()
-        elif isinstance(urng, np.random.Generator):
-            state = urng.bit_generator.state
-        else:
-            raise ValueError("Invalid random_state object")
-        x = self.rvs(size=size, random_state=urng)
-        # reset urng to get the same pseudo-uniform numbers used to create x
-        if isinstance(urng, np.random.RandomState):
-            urng.set_state(state)
-        elif isinstance(urng, np.random.Generator):
-            urng.bit_generator.state = state
+        # urng will be used to draw the samples for testing the error
+        # it must not interfere with self.random_state. therefore, do not
+        # call self.rvs, but draw uniform random numbers and apply
+        # self.ppf_fast (note: like in rvs, consider self._mirror_uniform)
```

Comment:
To me this sounds like an implementation detail. Why not just creating another rng, based on the provided rng, when the class is instantiated?

Reply:
What I mean is that IMO we should not ask the user to handle 2 RNG but do this ourselves internally. We can spawn another RNG based on the first one. i.e. we would have another private attribute for this RNG.

As for given the user the flexibility to update the value. I'm still not convinced about the need to be able to update the "main" RNG. Having the possibility to update this one is even more nebulous to me.

Reply:
for the evaluation of the error, it is important to leave the control to the user to make the calculation reproducible. this should not be a state in my view, here is really seems natural to say "estimate the error with seed XYZ"
we can take this up in the context of the other RNG topic :)